### PR TITLE
Improved tests for better test coverage

### PR DESCRIPTION
**Pull request by SatyendraBanjare**
Earlier the tests being run were not effectively covered. I have now modified and added a small coverage scripts for each module to do coverage analysis for that respective module. These scripts list functions only for that module. Now running tests modules differently does not alters coverage. The test run for any module will get a coverage analysis being done only for that module. Thus running tests for all modules, we should get coverage analysis of whole library's functions done.

### CODE REVIEW

**Comment by SatyendraBanjare** [comment]
Beautiful coverage result till now

```
==============================================================================
Summary
==============================================================================

File                                Hits Missed Coverage
--------------------------------------------------------
./annotated/annotated_cover.lua     14   1      93.33%
./annotated/test_annotated_spec.lua 38   0      100.00%
./blame/blame_cover.lua             18   1      94.74%
./blame/test_blame_spec.lua         34   0      100.00%
./blob/blob_cover.lua               21   2      91.30%
./blob/test_blob_spec.lua           46   0      100.00%
./branch/branch_cover.lua           29   2      93.55%
./branch/test_branch_spec.lua       59   0      100.00%
./checkout/checkout_cover.lua       11   0      100.00%
./checkout/test_checkout_spec.lua   39   0      100.00%
./commit/commit_cover.lua           50   3      94.34%
./commit/test_commit_spec.lua       138  0      100.00%
./config/config_cover.lua           78   7      91.76%
./config/test_config_spec.lua       163  0      100.00%
./cred/cred_cover.lua               20   3      86.96%
./cred/test_cred_spec.lua           23   0      100.00%
./describe/describe_cover.lua       11   0      100.00%
./describe/test_describe_spec.lua   31   0      100.00%
--------------------------------------------------------
Total                               823  19     97.74%

```

**Comment by SatyendraBanjare** [comment]
```
==============================================================================
Summary
==============================================================================

File                                  Hits Missed Coverage
----------------------------------------------------------
./annotated/annotated_cover.lua       14   1      93.33%
./annotated/test_annotated_spec.lua   38   0      100.00%
./blame/blame_cover.lua               18   1      94.74%
./blame/test_blame_spec.lua           34   0      100.00%
./blob/blob_cover.lua                 21   2      91.30%
./blob/test_blob_spec.lua             46   0      100.00%
./branch/branch_cover.lua             29   2      93.55%
./branch/test_branch_spec.lua         59   0      100.00%
./checkout/checkout_cover.lua         11   0      100.00%
./checkout/test_checkout_spec.lua     39   0      100.00%
./commit/commit_cover.lua             50   3      94.34%
./commit/test_commit_spec.lua         138  0      100.00%
./config/config_cover.lua             78   7      91.76%
./config/test_config_spec.lua         163  0      100.00%
./cred/cred_cover.lua                 20   3      86.96%
./cred/test_cred_spec.lua             23   0      100.00%
./describe/describe_cover.lua         11   0      100.00%
./describe/test_describe_spec.lua     31   0      100.00%
./diff/diff_cover.lua                 35   4      89.74%
./diff/test_diff_spec.lua             99   0      100.00%
./index/index_cover.lua               59   16     78.67%
./index/test_index_spec.lua           87   0      100.00%
./libgit/libgit_cover.lua             52   1      98.11%
./libgit/test_libgit_spec.lua         100  0      100.00%
./object/object_cover.lua             23   4      85.19%
./object/test_object_spec.lua         65   0      100.00%
./oid/oid_cover.lua                   23   2      92.00%
./oid/test_oid_spec.lua               71   0      100.00%
./reference/reference_cover.lua       56   15     78.87%
./reference/test_reference_spec.lua   105  0      100.00%
./reflog/reflog_cover.lua             25   0      100.00%
./reflog/test_reflog_spec.lua         57   0      100.00%
./repository/repository_cover.lua     70   13     84.34%
./repository/test_repository_spec.lua 103  0      100.00%
./reset/reset_cover.lua               5    0      100.00%
./reset/test_reset_spec.lua           24   0      100.00%
./revert/revert_cover.lua             5    0      100.00%
./revert/test_revert_spec.lua         20   0      100.00%
./revparse/revparse_cover.lua         7    0      100.00%
./revparse/test_revparse_spec.lua     33   0      100.00%
./revwalk/revwalk_cover.lua           18   1      94.74%
./revwalk/test_revwalk_spec.lua       38   0      100.00%
./signature/signature_cover.lua       13   0      100.00%
./signature/test_signature_spec.lua   41   0      100.00%
./tag/tag_cover.lua                   32   5      86.49%
./tag/test_tag_spec.lua               73   0      100.00%
./tree/test_tree_spec.lua             64   0      100.00%
./tree/tree_cover.lua                 34   5      87.18%
----------------------------------------------------------
Total                                 2260 85     96.38%

```

File: tests/annotated/annotated_cover.lua [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+end
+
+luagit2_annotated.annotated_commit_from_ref = annotated_commit_from_ref
+luagit2_annotated.annotated_commit_from_revspec = annotated_commit_from_revspec
+luagit2_annotated.annotated_commit_lookup = annotated_commit_lookup
+luagit2_annotated.annotated_commit_from_fetchhead = annotated_commit_from_fetchhead
+luagit2_annotated.annotated_commit_id = annotated_commit_id
+luagit2_annotated.annotated_commit_free = annotated_commit_free
+
+return luagit2_annotated
```
Comment (Etiene) [inline tests/annotated/annotated_cover.lua:47]: All this wrapping looks unnecessary. There are more straightforward ways of reassigning functions, but see comment below.

File: tests/annotated/test_annotated_spec.lua [hunk 2]
```diff
@@ -4,6 +4,7 @@ local fixer = require("Fixtures.fix_repo")
 
 describe(" Annotated Methods Tests ", function()
     local luagit2 = require("luagit2")
+    local luagit2_annotated = require("annotated.annotated_cover")
```
Comment (Etiene) [inline tests/annotated/test_annotated_spec.lua:7]: All this wrapping and renaming to get the coverage is not a good design decision. Developers often look into the tests to try to understand how a library is used and what they will find here is not a real usage, but instead this workaround to get the coverage %. I suggest instead to try to get a deeper understanding at how luacov works and see if it is possible to solve this in a different way. Or alternatively, leaving the coverage results out for the moment until a better solution is found.

**Comment by SatyendraBanjare** [comment]
@Etiene , Yes mam totally agreed. I myself check for the tests first for understanding things better in any library. So for the time anything good is figured out, let's keep both the branches master and this one `test_coverage` active. The coverage analysis let be done using this branch only and the master one can have complete tests all in a normal way without any extra wrapping.

**Pull request closed without merging**